src/odemis/driver/smaract.py

```diff
@@ -2931,6 +2956,16 @@ def _doReference(self, future, axes):
                         logging.warning("Axis %s not referenced after the end of referencing", a)
                         # TODO: Raise some error here
 
+                    # if referenced, move to the safe posiiton (if defined)
+                    if self._pos_deactive_after_ref and self._deactive_pos_after_ref is not None \
+                        and a in self._deactive_pos_after_ref:
+                        deactive_pos = self._deactive_pos_after_ref[a]
+                        logging.info("Moving axis %s to deactivated position %f", a, deactive_pos)
+                        f = self._createMoveFuture()
+                        self._doMoveAbs(f, {a: deactive_pos})
```

Comment:
As it is, it's going to move each axis serially. It's quite slow, and will have a strange global movement if there are several axes. So I'd recommend to only call _doMoveAbs(), with all the axes which should be moved. => Copy the deactivate pos, with only the axes present in the referencing.